File: Template/Naked Functions Server (F#)/Template.Model.Fsharp/ExampleDbContext.fs
﻿namespace Template.Model.FSharp

open Microsoft.EntityFrameworkCore
open Template.Model.Fsharp.Types

module Db =

    type ExampleDbContext(options) =
        inherit DbContext(options)

        [<DefaultValue>]
        val mutable _Students: DbSet<Student>

        member this.Students
            with public get () = this._Students
            and public set value = this._Students <- value

        override this.OnConfiguring(optionsBuilder: DbContextOptionsBuilder) =
            ()

        override this.OnModelCreating(modelBuilder: ModelBuilder) =
            modelBuilder
                .Entity<Student>()
                .HasData(new Student(Id = 1, FullName = "Alie Algol"))
            |> ignore

            modelBuilder
                .Entity<Student>()
                .HasData(new Student(Id = 2, FullName = "Forrest Fortran"))
            |> ignore

            modelBuilder
                .Entity<Student>()
                .HasData(new Student(Id = 3, FullName = "James Java"))
            |> ignore

            ()

        member this.Delete() = this.Database.EnsureDeleted()

        member this.Create() = this.Database.EnsureCreated()

File: Template/Naked Functions Server (C#)/Template.Model/ExampleDbContext.cs
using Microsoft.EntityFrameworkCore;
using Template.Model.Types;

namespace Template.Model;

public class ExampleDbContext : DbContext {
    public ExampleDbContext(DbContextOptions<ExampleDbContext> options)
        : base(options) { }

    public DbSet<Student> Students { get; set; }

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) { }

    protected override void OnModelCreating(ModelBuilder modelBuilder) {
        modelBuilder.Entity<Student>().HasData(new Student { Id = 1, FullName = "Alie Algol" });
        modelBuilder.Entity<Student>().HasData(new Student { Id = 2, FullName = "Forrest Fortran" });
        modelBuilder.Entity<Student>().HasData(new Student { Id = 3, FullName = "James Java" });
    }

    public void Delete() => Database.EnsureDeleted();

    public void Create() => Database.EnsureCreated();
}